conf/sage.config

```diff
@@ -0,0 +1,58 @@
+params {
+  config_profile_description = 'The Sage Bionetworks profile'
+  config_profile_contact = 'Bruno Grande (@BrunoGrandePhD)'
+  config_profile_url = 'https://github.com/Sage-Bionetworks-Workflows'
+}
+
+process {
+
+  cpus   = { check_max( 1    * slow(task.attempt), 'cpus'   ) }
+  memory = { check_max( 6.GB * task.attempt,       'memory' ) }
+  time   = { check_max( 24.h * task.attempt,       'time'   ) }
+
+  errorStrategy = { task.exitStatus in [143,137,104,134,139,247] ? 'retry' : 'finish' }
+  maxRetries    = 5
+  maxErrors     = '-1'
+
+  // Process-specific resource requirements
+  withLabel:process_low {
+    cpus   = { check_max( 4     * slow(task.attempt),  'cpus'   ) }
+    memory = { check_max( 12.GB * task.attempt,        'memory' ) }
+    time   = { check_max( 24.h  * task.attempt,        'time'   ) }
+  }
+  withLabel:process_medium {
+    cpus   = { check_max( 12    * slow(task.attempt), 'cpus'   ) }
+    memory = { check_max( 36.GB * task.attempt,       'memory' ) }
+    time   = { check_max( 48.h  * task.attempt,       'time'   ) }
+  }
+  withLabel:process_high {
+    cpus   = { check_max( 24    * slow(task.attempt), 'cpus'   ) }
+    memory = { check_max( 72.GB * task.attempt,       'memory' ) }
+    time   = { check_max( 96.h  * task.attempt,       'time'   ) }
+  }
+  withLabel:process_long {
+    time   = { check_max( 192.h  * task.attempt,   'time'   ) }
+  }
+  withLabel:process_high_memory {
+    memory = { check_max( 200.GB * task.attempt,   'memory' ) }
+  }
+
+}
```

Comment:
I'm not sure that there's necessarily a positive correlation between memory requirements and expected runtime (for example tools/scripts that use only one cpu and only a small amount of RAM but run for a long time), but I don't have a problem with this since these are always customizable by the user

Reply:
I totally agree, but this is the general model that nf-core uses ([example](https://github.com/nf-core/rnaseq/blob/master/conf/base.config#L21-L42)). They label processes with one of these to generally capture the "resource intensity", but I would like to consider memory- and CPU-intensive processes separately so we can optimize resource allocation. 

In general, I've been finding that our workflows aren't using all of the CPUs at their disposal, so I increased the CPU-to-memory ratio for these general labels. I also plan to create pipeline-specific config files once I delve into the resource stats of actual workflow runs, which I can add to the `sage` profile.